optimum/intel/neural_compressor/trainer.py

```diff
@@ -549,16 +582,13 @@ def compute_loss(self, model, inputs, return_outputs=False):
             # We don't use .loss here since the model may return tuples instead of ModelOutput.
             loss = outputs["loss"] if isinstance(outputs, dict) else outputs[0]
 
-        if (
-            self.is_in_train
-            and hasattr(self, "agent")
-            and hasattr(self.agent, "criterion")
-            and self.agent.criterion is not None
-        ):
+        if (self.is_in_train and hasattr(self, "agent") and hasattr(self.agent, "criterion") \
+            and self.agent.criterion is not None):
             logits = self._get_logits(outputs)
             teacher_logits = inputs.pop("teacher_logits", None)
-            # Compute teacher model outputs
-            if teacher_logits is None:
+            if teacher_logits is not None:
+                teacher_logits = tuple(teacher_logits.transpose(1, 0))

```

Comment:
this way of dealing teacher_logits only suit for SQuAD task, will be wrong for GLUE tasks since they usually have only one logit instead of two logits like SQuAD.

Reply:
I have fixed it by conditional statements. The operation of transpose will be implemented only if the task has two logits.